internal/log/log_test.go

```diff
@@ -0,0 +1,202 @@
+package log
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"log/slog"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/dhis2-sre/im-manager/internal/middleware"
+	"github.com/dhis2-sre/im-manager/pkg/model"
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLogs(t *testing.T) {
+	gin.SetMode(gin.TestMode)
+
+	r := gin.New()
+	r.Use(middleware.RequestID())
+
+	var b bytes.Buffer
+	logger := slog.New(New(slog.NewJSONHandler(&b, nil)))
```

Comment:
We are using the `TextHandler` right now in main. We are using the `JSONHandler` in the test so we can parse the logs using the `encoding/json` package. This makes it easy to assert on keys and values.

Since `slog.Handler` is an interface it does not really matter which implementation we use in a test vs in production.